cmd/namespace-lock.go

```diff
@@ -42,15 +45,21 @@ func initDsyncNodes() error {
 	// Initialize rpc lock client information only if this instance is a distributed setup.
 	clnts := make([]dsync.NetLocker, len(globalEndpoints))
 	myNode := -1
+	localLockServerIndex := 0
 	for index, endpoint := range globalEndpoints {
-		clnts[index] = newLockRPCClient(authConfig{
-			accessKey:       cred.AccessKey,
-			secretKey:       cred.SecretKey,
-			serverAddr:      endpoint.Host,
-			secureConn:      globalIsSSL,
-			serviceEndpoint: pathutil.Join(minioReservedBucketPath, lockServicePath, endpoint.Path),
-			serviceName:     lockServiceName,
-		})
+		if !endpoint.IsLocal {
+			clnts[index] = newLockRPCClient(authConfig{
+				accessKey:       cred.AccessKey,
+				secretKey:       cred.SecretKey,
+				serverAddr:      endpoint.Host,
+				secureConn:      globalIsSSL,
+				serviceEndpoint: pathutil.Join(minioReservedBucketPath, lockServicePath, endpoint.Path),
+				serviceName:     lockServiceName,
+			})
+		} else {
+			clnts[index] = &(globalLockServers[localLockServerIndex].ll)
```

Comment:
I really didn't understand the purpose of `localLockServerIndex`, don't we have only one local ?